cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/services/ApplicationLifecycleService.java

```diff
@@ -433,9 +433,13 @@ public void createAppDetailsArchive(ZipOutputStream zipOut) {
    * @return a collection of verion strings
    */
   public Collection<String> getAppVersions(NamespaceId namespaceId, String application) {
-    return store.getAllAppVersionsAppIds(namespaceId.app(application)).stream()
-      .map(ApplicationId::getVersion)
-      .collect(Collectors.toList());
+    List<String> appVersions = store.getAllAppVersionsAppIds(namespaceId.app(application)).stream()
+      .map(ApplicationId::getVersion).collect(Collectors.toList());
+    // Hide -SNAPSHOT when more than one version of the app exists
+    if (appVersions.size() > 1 && appVersions.contains("-SNAPSHOT")) {
```

Comment:
instead of putting a hardcode string, we can use ApplicationId.DEFAULT_VERSION

Reply:
good point, updated